feg/gateway/services/n7_n40_proxy/servicers/create_session_test.go

```diff
@@ -0,0 +1,336 @@
+		mock.MatchedBy(matchSmPolicyContextData(IMSI1_NOPREFIX)),
+	).Return(&n7_sbi.PostSmPoliciesResponse{HTTPResponse: &http.Response{StatusCode: 400}}, nil).Once()
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.Error(t, err)
+	mockN7.AssertExpectations(t)
+	assert.Nil(t, response)
+}
+
+func TestDisableN7Response(t *testing.T) {
+	srv, mockDb, mockN7 := newCentralSessionControllerForTest(t, true)
+
+	mockDb.On("GetOmnipresentRules").Return(
+		[]string{"omni_rule_1", "omni_rule_2"}, []string{"base_10"}).Once()
+	mockDb.On("GetRuleIDsForBaseNames", []string{"base_10"}).Return([]string{"base_rule_1", "base_rule_2"})
+
+	csr := newCreateSessionRequest(IMSI1)
+	response, err := srv.CreateSession(context.Background(), csr)
+	require.NoError(t, err)
+	mockDb.AssertExpectations(t)
+	mockN7.AssertNotCalled(t, "PostSmPoliciesWithResponse")
+
+	assert.Equal(t, genSessionID(IMSI1), response.SessionId)
+	allRuleIds := []string{}
+	for _, staticRule := range response.StaticRules {
+		allRuleIds = append(allRuleIds, staticRule.RuleId)
+	}
+	assert.ElementsMatch(t, []string{"omni_rule_1", "omni_rule_2", "base_rule_1", "base_rule_2"}, allRuleIds)
+}
+
+func newCentralSessionControllerForTest(t *testing.T, disableN7 bool) (*servicers.CentralSessionController, *mockPolicyDB.PolicyDBClient, *mockN7.ClientWithResponsesInterface) {
+	testN7Conf := getTestN7Config(t)
+	testN7Conf.DisableN7 = disableN7
+	mockPolicyDBClient := &mockPolicyDB.PolicyDBClient{}
+	mockPolicyClient := &mockN7.ClientWithResponsesInterface{}
+
+	srv, err := servicers.NewCentralSessionController(testN7Conf, mockPolicyDBClient, mockPolicyClient)
+	require.NoError(t, err)
+	return srv, mockPolicyDBClient, mockPolicyClient
+}
+
+func getTestN7Config(t *testing.T) *n7.N7Config {
+	apiRoot, err := url.ParseRequestURI(URL1)
+	require.NoError(t, err)
+	return &n7.N7Config{
+		DisableN7: false,
+		Server: n7.PCFConfig{
+			ApiRoot:      *apiRoot,
+			TokenUrl:     TOKEN_URL,
+			ClientId:     CLIENT_ID,
+			ClientSecret: CLIENT_SECRET,
+		},
+		Client: n7.N7ClientConfig{
+			LocalAddr:     LOCAL_ADDR,
+			NotifyApiRoot: NOTIFY_API_ROOT,
+		},
+	}
+}
+
+func newCreateSessionRequest(imsi string) *protos.CreateSessionRequest {
+	return &protos.CreateSessionRequest{
+		CommonContext: &protos.CommonSessionContext{
+			Sid:     &protos.SubscriberID{Id: imsi},
+			RatType: protos.RATType_TGPP_NR,
+			UeIpv4:  UE_IPV4,
+			Apn:     APN1,
+		},
+		RatSpecificContext: &protos.RatSpecificContext{
+			Context: &protos.RatSpecificContext_M5GsmSessionContext{
+				M5GsmSessionContext: &protos.M5GSMSessionContext{
+					PduSessionId:   uint32(PDU_SESSION_ID),
+					Gpsi:           GPSI1,
+					PduSessionType: protos.PduSessionType_IPV4,
+				},
+			},
+		},
+		SessionId:      genSessionID(IMSI1),
+		AccessTimezone: &protos.Timezone{OffsetMinutes: 330},
+	}
+}
+
+func genSessionID(imsi string) string {
+	return fmt.Sprintf("%s-1234", imsi)
+}
```

Comment:
i don't think you need this. Use a static value

Reply:
This is required in some places where imsi is passed as an argument.

Reply:
Just define a variable on the top of the file with that value

```
sessionID1 :=  fmt.Sprintf("%s-1234", IMSI1)
sessionID1 :=  fmt.Sprintf("%s-1234", IMSI2)
```

The reason for that is when debugging a test it is easier to see the how the values were generated.

Also in test is better to be verbose some times. Meaning, doing this on a regular file would be bad, but on a test may be helpful because is easier to understand for debugging

Reply:
Done